commandsv3/src/main/java/org/wpilib/commands3/Scheduler.java

```diff
@@ -0,0 +1,891 @@
+   * scheduler cycle per nesting level; with the default 20ms update period, 5 levels of nesting
+   * would be enough to delay actions by 100 milliseconds - instead of only 20.
+   *
+   * <p>Does nothing if the command is already scheduled or running, or requires at least one
+   * mechanism already used by a higher priority command.
+   *
+   * <p>If one command schedules another ("parent" and "fork"), the forked command will be canceled
+   * when the parent command completes. It is not possible to fork a command and have it live longer
+   * than the command that forked it.
+   *
+   * @param command the command to schedule
+   * @return the result of the scheduling attempt. See {@link ScheduleResult} for details.
+   * @throws IllegalArgumentException if scheduled by a command composition that has already
+   *     scheduled another command that shares at least one required mechanism
+   */
+  public ScheduleResult schedule(Command command) {
+    // Note: we use a throwable here instead of Thread.currentThread().getStackTrace() for easier
+    //       stack frame filtering and modification.
+    var binding =
+        new Binding(
+            BindingScope.global(), BindingType.IMMEDIATE, command, new Throwable().getStackTrace());
+
+    return schedule(binding);
+  }
+
+  // Package-private for use by Trigger
+  ScheduleResult schedule(Binding binding) {
+    var command = binding.command();
+
+    if (isScheduledOrRunning(command)) {
+      return ScheduleResult.ALREADY_RUNNING;
+    }
+
+    if (!isSchedulable(command)) {
+      return ScheduleResult.LOWER_PRIORITY_THAN_RUNNING_COMMAND;
+    }
+
+    for (var scheduledState : m_onDeck) {
+      if (!command.conflictsWith(scheduledState.command())) {
+        // No shared requirements, skip
+        continue;
+      }
+      if (command.isLowerPriorityThan(scheduledState.command())) {
+        // Lower priority than an already-scheduled (but not yet running) command that requires at
+        // one of the same mechanism. Ignore it.
+        return ScheduleResult.LOWER_PRIORITY_THAN_RUNNING_COMMAND;
+      }
+    }
+
+    // Evict conflicting on-deck commands
+    // We check above if the input command is lower priority than any of these,
+    // so at this point we're guaranteed to be >= priority than anything already on deck
+    evictConflictingOnDeckCommands(command);
+
+    // If the binding was declared inside a running command, that command is the parent (the current
+    // command will always be null in this case, since triggers are polled before commands are
+    // mounted and run). Otherwise, the parent is the command that's running at the time `schedule`
+    // is called - which may be null.
+    var parent =
+        binding.scope() instanceof BindingScope.ForCommand scope
+            ? scope.command()
+            : currentCommand();
+    var state = new CommandState(command, parent, buildCoroutine(command), binding);
+
+    emitEvent(scheduled(command));
```

Comment:
`scheduled` here is very confusing. Rename to `sceduledEvent` or something

Reply:
Agreed. I'll simplify these to eg `emitScheduledEvent(...)` etc